opencog/haskell/TruthValue_CWrapper.cpp

```diff
@@ -0,0 +1,107 @@
+#include "TruthValue_CWrapper.h"
+
+
+int TruthValue_toRawType(TruthValuePtr tv
+                        , char** tv_type
+                        , double* parameters)
+{
+    const std::string & tvt = classserver().getTypeName(tv->getType());
+
+    *tv_type = (char*) malloc(sizeof(char) * (tvt.length()+1));
+    if(! *tv_type)
+        throw RuntimeException(TRACE_INFO,"Failed malloc.");
+    std::strcpy(*tv_type, tvt.c_str());
+
+    if (strcmp(*tv_type,"SimpleTruthValue") == 0){
```

Comment:
A simpler solution would have been to just cast these to a FloatValue, which already defines an array of floats that occur in exactly the correct order.  Here, you are confusingly defining an array that just might happen to be in a different order from the "natural" order of the truth value.

Also, given that all truth values are now arrays, it would simplify the code -- you would not need any kind of special-casing here, for each of the different types.